libevmasm/Assembly.cpp

```diff
@@ -495,16 +519,36 @@ LinkerObject const& Assembly::assemble() const
 	// Otherwise ensure the object is actually clear.
 	assertThrow(m_assembledObject.linkReferences.empty(), AssemblyException, "Unexpected link references.");
 
+	LinkerObject& ret = m_assembledObject;
+
 	size_t subTagSize = 1;
 	for (auto const& sub: m_subs)
 	{
-		sub->assemble();
+		auto const& linkerObject = sub->assemble();
```

Comment:
This just copies the references from the sub-objects without adding any offset and without storing which sub-object it comes from. Need to investigate if this is fine.